c10/core/TensorImpl.h

```diff
@@ -657,6 +657,13 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
     return storage_;
   }
 
+  inline StorageImpl* unsafeMaybeGetStorageImpl() const {
+    if (C10_UNLIKELY(storage_access_should_throw_)) {
+      return nullptr;
+    }
+    return storage().unsafeGetStorageImpl();
```

Comment:
It turns out some of the new structured kernels are used for `CompositeExplicitAutograd` which means they build a `TensorIterator` with potentially non-strided tensors types. That means there is a strict need to support `TensorImpl` subclasses. 

However, for our purposes I think we only really care about normal strided tensors. So, would it be fine to just do `storage_ ? storage_.unsafeGetStorageImpl() : nullptr`?  I have a hard time thinking of a use case where `storage_` would not be populated but `storage()` would return something sensible.

Reply:
Sorry about the late response. I agree with the diagnosis here: you can assume CompositeExplicitAutograd only happens for strided kernels. We could make this a little safer by introducing a separate alias key for strided keys only.